### DISCUSSION TITLE
Enhancement to route builder: graph of grade %

### DISCUSSION THREAD

**Discussion by bkonold** [SauceLLC/sauce4strava#162] [Ideas]
Hello,

One thing that I love about Ride with GPS is the ability to examine a route's grade profile very clearly with its own graph. You cannot quite do this on Strava - you can only see visually the steepness of the elevation curve, or hover over single points to see the grade percentage at that point. Example from Strava below.

<img width="500" alt="Screen Shot 2023-07-07 at 10 36 12 PM" src="https://github.com/SauceLLC/sauce4strava/assets/2802866/bfcae527-43df-40ff-9769-12390a46fc91">

If the data is accessible to a chrome extension, it would be wonderful to have a view like Ride with GPS does, as it makes it much easier to gauge the real difficulty of a route (or to develop a pacing strategy). Below is an example from their website.

<img width="500" alt="Screen Shot 2023-07-07 at 10 29 22 PM" src="https://github.com/SauceLLC/sauce4strava/assets/2802866/ed14c98e-58cf-4bd1-a6c9-a45361a9c22d">

**Comment by mayfield** [comment]
Unfortunately Route Builder is a `React` based page and as such is completely non-extendable.  Kinda sad how popular that framework is TBH.
